### ISSUE TITLE
ToolTip on TransactionModel is misleading

### ISSUE THREAD

**Issue by lvanvugt** [Microsoft/AL#3819]
As the title describes: the ToolTip on TransactionModel is misleading.

Creating a new test function and wanting to set the TransactionModel the ToolTip shown in VSCode tells me:

`TransactionModel(TransactionModel: TransactionModel)`

Filling in, for example, AutoRollback after the colon will show an error:

`TransactionModel(TransactionModel: AutoRollback )`

So I tried the scope operator ::

`TransactionModel(TransactionModel::AutoRollback )`

And this is fine. SO the Tooltip has to be changed to

`TransactionModel(TransactionModel::TransactionModel)`

**Comment by qutreson** [comment]
Hi @lvanvugt, this is actually by design and works as inteded. The syntax used on hover is the following one:
```
AttributeWithSignature(AttributeArgumentName: AttributeArgumentType)
```
We always show the type and the name of the expected attribute arguments and not how to declare them. This also follows the syntax used by signature help.

For instance, when declaring an EventSubscriber, the tool tip is also as follows:
```
EventSubscriber(ObjectType: ObjectType, ObjectId: Integer, EventName: Text, ElementName: Text, SkipOnMissingLicense: Boolean, SkipOnMissingPermission: Boolean)
```

Where the ObjectType also defines an option, for instance: `ObjectType::Page`.
As the TransactionModel attribute expects one argument named TransationModel of type TransactionModel, the signature is as follows:

```
TransactionModel(TransactionModel: TransactionModel)
```

**Comment by lvanvugt** [comment]
Hi @qutreson makes snese, but it's a darn job to find out what to exactly provide/fill in and intellisense is of no help here.

**Comment by qutreson** [comment]
I'll use this issue in order to track later improvement for IntelliSense when suggesting options. 

Probably we could include the scope operator `::` when selecting the option `TransactionModel` among the suggestions for the attribute argument. So that you would directly get `TransactionModel(TransactionModel::` and you would then only need to trigger IntelliSense a second time in order to choose which value you want to use.

**Comment by lvanvugt** [comment]
Agree. And ... intellisense should already show me TransactionModel as the (only) possible value for the property.

**Comment by lvanvugt** [comment]
What's the status on this right now?
